### ISSUE TITLE
[New Data Source] Apache Pinot

### ISSUE THREAD

**Issue by sunny19930321** [grafana/grafana#20141]
Apache Pinot is a realtime distributed OLAP datastore, which is used at LinkedIn, Uber, Microsoft, Slack and Weibo to deliver scalable real time analytics with low latency. for more details: https://pinot.apache.org/

**Comment by torkelo** [comment]
@ryantxu https://pinot.apache.org/ looks interesting :)

**Comment by aned** [comment]
Indeed, would be nice!

**Comment by ryantxu** [comment]
https://github.com/fx19880617/pinot-client-go

Maybe

**Comment by HunterEl** [comment]
Any updates on this?

**Comment by ryantxu** [comment]
Interesting... now there is a go client, so should be possible:
https://github.com/fx19880617/pinot-client-go

**Comment by HunterEl** [comment]
@ryantxu agreed. I see `pinot` as a data-source in the coursera fork of grafana: https://github.com/coursera/grafana/tree/pinot-datasource/public/app/plugins/datasource/pinot granted this is 5 years old. Would be interesting to see how much of this is still usable and could be ported in. Happy to work on this if this is alright with maintainers!

**Comment by kishoreg** [comment]
@HunterEl let us know if you need any help with this.

**Comment by fx19880617** [comment]
Please let me know if there is anything I can help with.

**Comment by rmatharu** [comment]
Is graphana supported with Pinot now?

**Comment by wbrowne** [comment]
Looping in @grafana/enterprise-plugins in case this is something we would want on our radar.

Alternatively, if anyone feels like building a datasource for this then [this](https://grafana.com/docs/grafana/latest/developers/plugins/backend/) is a good starting point.

**Comment by ajameson-hbo** [comment]
Does anyone know if there's recent effort on this plugin be it open source or enterprise?

**Comment by bagipriyank** [comment]
> Hi Any update one this? We are on pinot production, any one wana put joint effort to build this?

I can help. We are also looking into querying Pinot via Grafana. I think for now we will use Trino in between but would like to get rid of the unnecessary dependency.

**Comment by bagipriyank** [comment]
> Looping in @grafana/enterprise-plugins in case this is something we would want on our radar.
> 
> Alternatively, if anyone feels like building a datasource for this then [this](https://grafana.com/docs/grafana/latest/developers/plugins/backend/) is a good starting point.

I am guessing this is the new link https://grafana.com/developers/plugin-tools/tutorials/build-a-data-source-backend-plugin#anatomy-of-a-backend-plugin

**Comment by sympatheticmoose** [comment]
Added a new label for our teams to re-review as a potential datasource - @bagipriyank you're correct, https://grafana.com/developers/plugin-tools/tutorials/build-a-data-source-backend-plugin would get you started.

**Comment by edgarasg** [comment]
Related https://github.com/grafana/grafana/issues/90776

**Comment by PeterCorless** [comment]
There is a corresponding Apache Pinot request for a Grafana Plugin: [Support Pinot as a DataSource in Grafana plugin #6639](https://github.com/apache/pinot/issues/6639):
